src/shared/tests.h

```diff
@@ -251,7 +255,7 @@ static inline int run_test_table(void) {
         ({                                                                                                      \
                 const char* _expr1 = (expr1);                                                                   \
                 const char* _expr2 = (expr2);                                                                   \
-                if (strcmp(_expr1, _expr2) != 0) {                                                              \
+                if (!streq_ptr(_expr1, _expr2) != 0) {                                                          \
```

Comment:
```suggestion
                if (!streq_ptr(_expr1, _expr2)) {                                                          \
```

Reply:
I have no idea how that managed to sneak in :)